src/Sustainsys.Saml2/Validation/AssertionValidator.cs

```diff
@@ -14,8 +14,15 @@ public void Validate(
     {
         ValidateIssuer(assertion, parameters);
         ValidateConditions(assertion.Conditions, parameters);
+        ValidateSubjectConfirmation(assertion, parameters);
+        ValidateSubjectConfirmationData(assertion, parameters);
```

Comment:
I think ValidateSubjectConfirmationData should be called from ValidateSubjectConfirmation as it is part of the SubjectConfirmation. We should go one step further and add a ValidateSubject that calls ValidateSubjectConfirmation that calls ValidateSubjectConfirmationData.

ValidateSubject should check that there is a Subject. It is mandatory in the WebSSO profile so we should require it. The methods should not accept Assertions but instead be
```
ValidateSubject(Subject subject)
ValidateSubjectConfirmation(SubjectConfirmation confirmation)
ValidateSubjectConfirmationData(SubjectConfirmationData confirmationData).
```

This will make null checking easier.